include/circt/Dialect/FIRRTL/FIRRTLAttributes.td

```diff
@@ -232,20 +232,28 @@ def InnerSymAttr : AttrDef<FIRRTLDialect, "InnerSym"> {
   let parameters = (ins ArrayRefParameter<"InnerSymPropertiesAttr">:$props);
   let builders = [
     AttrBuilderWithInferredContext<(ins "::mlir::StringAttr":$sym),[{
-      return get(sym.getContext(), 
+      if (!sym)
+        return InnerSymAttr();
```

Comment:
Can we remove the gate `if(!sym) return ..`? I think it is users' responsibility to guarantee that the given attribute is non-null within `AttrBuilderWithInferredContext`, e.g https://github.com/llvm/llvm-project/blob/26dcc7e64afe089c78884213ee7f56723c95cb80/mlir/include/mlir/IR/BuiltinAttributes.td#L387-L392

Reply:
Thanks @uenoku for this suggestion. Actually I think its better to remove this check. I updated the users to not rely on this check.